### PR TITLE
[SPARK-39400][SQL] spark-sql should remove hive resource dir in all case

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
In current code, when we use `spark-sql` `-e` , `-f` or use `ctrl + c` to close  `spark-sql` session, will remain hive session resource dir under `/tmp` path, this pr help to clean this files

### Why are the changes needed?
Clean remained files

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manuel tested

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan @wangyum

**Review by wangyum — COMMENTED** [review]
Could we add a test?

**Comment by AngersZhuuuu** [comment]
> Could we add a test?

Need to test after built, seems hard to write test in UT...

**Review by wangyum — APPROVED** [review]

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala [hunk 1]
```diff
@@ -140,7 +143,10 @@ private[hive] object SparkSQLCLIDriver extends Logging {
     SessionState.setCurrentSessionState(sessionState)
 
     // Clean up after we exit
-    ShutdownHookManager.addShutdownHook { () => SparkSQLEnv.stop() }
+    ShutdownHookManager.addShutdownHook { () =>
+      sessionState.close()
+      SparkSQLEnv.stop()
```
Comment (wangyum) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala:148]:
```scala
    try {
      sessionState.close()
    } finally {
      SparkSQLEnv.stop()
    }
```
?

Comment (AngersZhuuuu) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala:148]: `sessionState.close` already have try catch
```
  public void close() {
    try {
      super.close();
    } catch (IOException var2) {
      var2.printStackTrace();
    }

  }
```

**Review by cloud-fan — APPROVED** [review]

**Comment by wangyum** [comment]
Merged to master.

**Pull request closed without merging**